src/main/java/uk/nhs/hee/tis/trainee/credentials/service/RevocationService.java

```diff
@@ -89,23 +93,32 @@ public void revoke(String tisId, CredentialType credentialType) {
     List<CredentialMetadata> validCredentialsMetadata = credentialsMetadata.stream()
         .filter(meta -> meta.getRevokedAt() == null)
         .toList();
-
-    if (!validCredentialsMetadata.isEmpty()) {
-      log.info("{} Issued credential(s) of type {} found for TIS ID {}, revoking.",
-          validCredentialsMetadata.size(), credentialType, tisId);
-      validCredentialsMetadata.forEach(metadata -> {
-        gatewayService.revokeCredential(credentialType.getTemplateName(),
-            metadata.getCredentialId());
-
-        metadata.setRevokedAt(Instant.now());
-        credentialMetadataRepository.save(metadata);
-        eventPublishingService.publishRevocationEvent(metadata);
-        log.info("Credential {} for TIS ID {} has been revoked.", credentialType, tisId);
-      });
-
-    } else {
-      log.info("No {} credential issued for TIS ID {}, skipped revocation.", credentialType, tisId);
-    }
+      List<Integer> originalCredentialHashCodes = credentialsMetadata.stream()
+          .filter(meta -> meta.getRevokedAt() == null)
+          .map(CredentialMetadata::getCredentialHashCode)
+          .collect(Collectors.toList());
+
+      if(credentialHashCode != null && originalCredentialHashCodes.contains(credentialHashCode))
```

Comment:
This isn't quite right. As is it would skip revocation for the whole list if any of them match the hash, we'd still want to revoke any in the list that don't match.